src/main/perf/LineFileDocs.java

```diff
@@ -501,7 +503,20 @@ static Document cloneDoc(Document doc1) {
     return doc2;
   }
 
-  /* this function make sure the calling thread will have something to index */
+  /* Call this function to put the remaining doc block into recycle queue */
+  public void recycle() {
+    if (isBinary && nextDocs.get() != null && nextDocs.get().getBlockByteText().hasRemaining()) {
```

Comment:
This should never be called when `isBinary == true` right?  Maybe throw an exception in that case, or change this to `assert`?

Reply:
I think this will be called in both binary and txt case because we don't distinguish them outside this class, so I keep `isBinary` here to filter only the binary case